src/expanduser.cpp

```diff
@@ -35,23 +35,23 @@ MIOPEN_DECLARE_ENV_VAR(HOME)
 
 namespace miopen {
 
-std::string ExpandUser(const std::string& path)
+static std::string GetHomeDir()
 {
-    std::string home_dir;
     const char* const p = GetStringEnv(HOME{});
     if(!(p == nullptr || p == std::string("/") || p == std::string("")))
```

Comment:
Is this thing `p == std::string("/")` checking first symbol is '/'? This style is very cryptic. Instead of these non-portable checks, please check p == nullptr and use boost again to check if the path exists, is a directory and is writable.

Reply:
This code is not related to this PR. Please feel free to open an issue or PR if you think it's worth it.